tests/utils/test_string_utils.py

```diff
@@ -32,6 +32,7 @@ def test_str_to_bool_exception_2():
 
 
 def test_str_to_bool_default():
+    assert not str_to_bool("blah")
```

Comment:
I expect a ValueError here, not a None returned, similar to float:
```
>>> float("abc")
ValueError: could not convert string to float: 'abc'
```

Example: What happens if x = "blah" in the following case:
```python
if str_to_bool(x):
   print("True")
else:
   print("False")
```

Reply:
And looking even closer: a default of True does not make sense. The only case where the exception in line 25 is raised is for None.

Reply:
I was going for matching the pattern seen in dict.get() where you can provide a default to be returned instead of raising KeyError. The strict option is unnecessary so I will change it to default=None instead to match the dict implementation. So:

```
str_to_bool("blah") # ValueError
str_to_bool("blah", default=False) # False
```

Does this sound ok?

Reply:
`dict.get()` is already the specialized code, whereas `dict[]` is the normal case. I like your example: a value-error without explicit `default` and the explicit default otherwise.